python/cudf/cudf/_lib/datetime.pyx

```diff
@@ -62,82 +62,63 @@ def extract_datetime_component(Column col, object field):
     return result
 
 
-def ceil_datetime(Column col, object field):
+cdef libcudf_datetime.rounding_frequency _get_rounding_frequency(object freq):
+    cdef libcudf_datetime.rounding_frequency freq_val
+
+    # https://pandas.pydata.org/pandas-docs/version/0.25.0/reference/api/pandas.Timedelta.resolution.html
+    if freq == "D":
+        freq_val = libcudf_datetime.rounding_frequency.DAY
+    elif freq == "H":
+        freq_val = libcudf_datetime.rounding_frequency.HOUR
+    elif freq == "T" or freq == "min":
```

Comment:
```suggestion
    elif freq in ("T", "min"):
```

ditto below

Reply:
This was previously suggested by @vyasr — this change requires the GIL and can’t be easily done in a cdef function iirc.

Reply:
Yep, putting a link to that: https://github.com/rapidsai/cudf/pull/9571/files#r743999332

Reply:
Just tried it with this suggestion and it works! I guess remove the gill made this possible @bdice!